app/controllers/users_controller.rb

```diff
@@ -0,0 +1,56 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user_params = params.require(:user).permit([:first_name,:last_name,:email,
+      :password,:password_confirmation])
+    @user = User.new user_params
+    if @user.save
+      redirect_to root_path, notice: "Created User Account Successfully"
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find params[:id]
+  end
+
+  def update
+    user_params = params.require(:user).permit([:first_name,:last_name,:email])
+    @user = User.find params[:id]
+    if @user.update user_params
+      redirect_to root_path, notice: "Your Information has been updated"
+    else
+      flash[:alert] = "Your Information has not been saved"
+      render :edit
+  end
+  end
+
+  def change_password
+    @user = User.find params[:user_id]
+  end
+
+  def change_password_update
+    user_params = params.require(:user).permit([:password,:password_confirmation])
```

Comment:
I would also add a check for if old_password == password, to make sure the user has actually changed the password, and also check if password.blank?